src/Stratis.Bitcoin.Features.Consensus/ProvenBlockHeaders/IProvenBlockHeaderRepository.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using NBitcoin;
+
+namespace Stratis.Bitcoin.Features.Consensus.ProvenBlockHeaders
+{
+    /// <summary>
+    /// Interface for database <see cref="ProvenBlockHeader"></see> repository.
+    /// </summary>
+    public interface IProvenBlockHeaderRepository : IDisposable
+    {
+        /// <summary>Loads <see cref="ProvenBlockHeader"/> items from the database.</summary>
+        /// <param name="blockHash">BlockId to initial the database with.</param>
+        /// <param name="cancellationToken">Cancellation that allows aborting the operation at any stage.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task InitializeAsync(uint256 blockHash = null, CancellationToken cancellationToken = default(CancellationToken));
+
+        /// <summary> Retrieves the block hash of the current <see cref="ProvenBlockHeader"/> tip.</summary>
+        /// <param name="cancellationToken">Cancellation that allows aborting the operation at any stage.</param>
+        /// <returns>A <see cref="Task"/> representing the asynchronous operation.</returns>
+        Task<uint256> GetTipHashAsync(CancellationToken cancellationToken = default(CancellationToken));
```

Comment:
Is there a specific reason that we only save and retrieve the tip's hash here? I see in the store that the equivalent method returns the entire `ProvenBlockHeader`, which seems more useful overall. Maybe I'm misunderstanding the differences in purpose between the repo and store.

Reply:
I wasn't too sure with this implementation.  

The table could only store the block hash tip of the last `ProvenBlockHeader` saved to disk.  Then the table will only contain one row.

Or we could include the `BlockHash` as a property on `ProvenBlockHeader`.  Then remove the BlockHash table.  I'm not sure of the ramifications here.  Will the BlockHash be available during the construction of a `ProvenBlockHeader`, for example.

@zeptin, @dangershony, @noescape00, @Aprogiena what do you think ?